torch/accelerator/__init__.py

```diff
@@ -0,0 +1,132 @@
+# mypy: allow-untyped-defs
+r"""
+This package introduces support for the current :ref:`accelerator<accelerators>` in python.
+"""
+
+from typing import Any
+
+import torch
+
+from ._utils import _device_t, _get_device_index
+
+
+def current_accelerator() -> torch.device:
+    r"""Return the device type of the current :ref:`accelerator<accelerators>`.
+
+    Returns:
+        torch.device: return the current accelerator as :class:`torch.device`.
+            If no available accelerators, return cpu device.
+
+    Example::
+
+        >>> # xdoctest: +SKIP
+        >>> if torch.accelerator.current_accelerator().type == 'cuda':
+        >>>     is_half_supported = torch.cuda.has_half()
+        >>> elif torch.accelerator.current_accelerator().type == 'xpu':
+        >>>     is_half_supported = torch.xpu.get_device_properties().has_fp16
+    """
+    return torch._C._accelerator_getAccelerator()
+
+
+def device_count() -> int:
+    r"""Return the number of current :ref:`accelerator<accelerators>` available.
+
+    Returns:
+        int: the number of the current :ref:`accelerator<accelerators>` available.
+            If no available accelerators, return 0.
+    """
+    return torch._C._accelerator_deviceCount()
+
+
+def is_available() -> bool:
+    r"""Check if there is an available :ref:`accelerator<accelerators>`.
+
+    Returns:
+        bool: A boolean indicating if there is an available :ref:`accelerator<accelerators>`.
+
+    Example::
+
+        >>> assert torch.accelerator.is_available() "No available accelerators detected."
+    """
+    return device_count() > 0
+
+
+def current_device_idx() -> int:
+    r"""Return the index of a currently selected device for the current :ref:`accelerator<accelerators>`.
+
+    Returns:
+        int: the index of a currently selected device.
+    """
+    return torch._C._accelerator_getDeviceIndex()
+
+
+def set_device(device: _device_t) -> None:
+    r"""Set the current device to a given device.
+
+    Args:
+        device (:class:`torch.device`, str, int): a given device that must match the current
+            :ref:`accelerator<accelerators>` device type. This function is a no-op if this argument is negative.
+    """
+    device_index = _get_device_index(device)
+    torch._C._accelerator_setDevice(device_index)
+
+
+def current_stream(device: _device_t = None) -> torch.Stream:
```

Comment:
For this and all other APIs, we are now 3.8+ meaning that we can use positional-only argument.
`def set_device(device: _device_t, /) -> None:` might be better here if we want to say that "device" as the argument name should not matter.

Especially because these can be either `device=torch.device(...)` or `device=2`. And the latter feels sooo wrong :D 

What do you think?

Reply:
It think It is a good point. Although `device=2` is legal and valid to construct `torch.device` with the current accelerator type and `index=2`, it still looks quite weird.